service/controller/machinepool/handler/machinepoolupgrade/lastdeployedreleaseversion.go

```diff
@@ -136,10 +136,10 @@ func allNodePoolNodesUpToDate(ctx context.Context, tenantClusterClient client.Cl
 
 	// azure-admission-controller ensures that machinePool.Spec.Replicas is
 	// always set
-	requiredReplicas := *machinePool.Spec.Replicas
+	minRequiredReplicas := key.NodePoolMinReplicas(machinePool)
```

Comment:
Can cluster-autoscaler be completely not enabled, so that these annotations are missing (or whatever is the behavior when it's generally disabled, not only during upgrades)? In that case it would fall back to `machinePool.Spec.Replicas`.

Reply:
ensured by the azure admission controller https://github.com/giantswarm/azure-admission-controller/blob/e0695e1d24a783c9dab971efff8d4490fa55c29e/pkg/machinepool/autoscaling.go#L24